src/hooks/useTruncate.tsx

```diff
@@ -0,0 +1,80 @@
+'use client';
```

Comment:
what does this do? SSR related?

Reply:
Yes, ensures this is only run on client as it uses DOM.